File: Python/Warmup-2/array_front9.py
"""
Dada um array numérico "nums", retorne true se um dos 4 elementos iniciais for um "9".
O "length" do array pode ser menor que 4.
"""


def array_front9(nums):
    return nums[:4].count(9) >= 1


# OUTRO JEITO MAIS EXPLICADO E PASSO A PASSO
def array_front9_2(nums):
    # First figure the end for the loop
    end = len(nums)
    if end > 4:
        end = 4

    for i in range(end):  # loop over index [0, 1, 2, 3]
        if nums[i] == 9:
            return True
    return False


print(array_front9([0, 1, 2, 9, 9]))

File: Java/Warmup-2/arrayFront9/ArrayFront9.java
package arrayFront9;

public class ArrayFront9 {
    //Retorna true se um dos 4 primeiros números do Array for 9, podendo o Array ter menos de 4 elementos.

    public boolean arrayFront9(int[] nums) {
        int size = 4;
        if (nums.length < 4){
            size = nums.length;
        }
        for (int i = 0; i < size; i++) {
            if (nums[i] == 9) return true;
        }
        return false;
    }
}
